implementations/micrometer-registry-otlp/src/test/java/io/micrometer/registry/otlp/OTelCollectorIntegrationTest.java

```diff
@@ -46,12 +48,25 @@ class OTelCollectorIntegrationTest {
 
     private static final String CONFIG_FILE_NAME = "collector-config.yml";
 
+    private static final DockerImageName COLLECTOR_IMAGE = DockerImageName
+        .parse("otel/opentelemetry-collector-contrib:" + getCollectorImageVersion());
+
     @Container
-    private final GenericContainer<?> container = new GenericContainer("otel/opentelemetry-collector-contrib")
+    private final GenericContainer<?> container = new GenericContainer(COLLECTOR_IMAGE)
         .withCommand("--config=/etc/" + CONFIG_FILE_NAME)
         .withClasspathResourceMapping(CONFIG_FILE_NAME, "/etc/" + CONFIG_FILE_NAME, READ_ONLY)
+        .waitingFor(Wait.forLogMessage(".*Everything is ready.*", 1))
         .withExposedPorts(4318, 9090); // HTTP receiver, Prometheus exporter
 
+    private static String getCollectorImageVersion() {
+        String version = System.getProperty("otel-collector-image.version");
+        if (version == null) {
+            throw new IllegalStateException(
+                    "System property 'otel-collector-image.version' is not set. This should be set in the build configuration for running from the command line. If you are running OTelCollectorIntegrationTest from an IDE, set the system property to the desired collector image version.");
```

Comment:
This error message is intended to make it obvious what to do for anyone who tries to run the test in the IDE for the first time, which will fail (unlike before this change).